### PR TITLE
[CP] Fix the incorrect ring schedule in the fwd and bwd

### PR DESCRIPTION
**Pull request by fegin**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129517
* #129516
* #130527
* #129515
* __->__ #129514
* #129512

Summary:
1. The argument order for all_to_all_single is "block, output_split_size, input_split_sizes, pg".
2. Uses the correct ring order for the grad_kv.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @tianyu-l @yf225 @chauhang

### CODE REVIEW

**Review by d4l3k — APPROVED** [review]
changes seem reasonable -- can we add a test to catch this? from what I recall before I had validated the output/gradient with the chunked single host version

**Review by drisspg — APPROVED** [review]

File: torch/distributed/_tensor/experimental/attention.py [hunk 1]
```diff
@@ -166,9 +166,7 @@ def _ring_rotate(block: torch.Tensor, pg: dist.ProcessGroup) -> torch.Tensor:
     output_split_sizes = [0] * size
     output_split_sizes[(rank - 1) % size] = len(block)
 
-    out = ft_c.all_to_all_single_autograd(
-        block, input_split_sizes, output_split_sizes, pg
-    )
+    out = ft_c.all_to_all_single(block, output_split_sizes, input_split_sizes, pg)
```
Comment (wanchaol) [inline torch/distributed/_tensor/experimental/attention.py:169]: I wonder if we could add some test case to guard the CP correctness?

**Review by wanchaol — APPROVED** [review]

**Comment by fegin** [comment]
I'm going to add the unittest to verify the correctness of CP. The verification needs some design though because of the result won't be identical to the non-CP SDPA.

**Pull request closed without merging**